Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -2200,26 +2194,11 @@ private void OnFocusChanged(bool focus)
 
         private bool ShouldRunUpdate(InputUpdateType updateType)
         {
-            switch (updateType)
-            {
-                case InputUpdateType.BeforeRender:
-                case InputUpdateType.Dynamic:
-                case InputUpdateType.Fixed:
-                    // Note: When Touchscreen Keyboard is active, Unity application looses focus, thus none of input is being processed
-                    //       Force input updating while keyboard is show
-                    //       In the future, hopefully we'll have TouchscreenKeyboard integrated in thew new input system directly
-                    //       Thus with removal of KeyboardOnScreen, KeyboardOnScreen::IsVisible() check should go away
-                    if (!(m_Settings.runInBackground || TouchScreenKeyboard.visible) && !m_HasFocus)
-                        return false;
-                    break;
 #if UNITY_EDITOR
-                case InputUpdateType.Editor:
-                    // If we're in play mode and the player has focus (or ignores focus), don't run editor updates.
-                    if (Application.isPlaying && !EditorApplication.isPaused && (m_HasFocus || m_Settings.runInBackground))
-                        return false;
-                    break;
+            // If we're in play mode and the player has focus (or ignores focus), don't run editor updates.
+            if (updateType == InputUpdateType.Editor && Application.isPlaying && !EditorApplication.isPaused)
+                return false;
 #endif
-            }
```

Comment:
This seems to be breaking focus behavior. For the player it should be fine but for the editor it would mean that the game view is eating input targeted at the editor.

Reply:
I don't understand. The behavior is the same as before _if runInBackground was enabled_? Should we maybe always run editor Updates?

Reply:
Think I misread what m_HasFocus *actually* means. However, that makes me wonder whether we had a regression here before. The code in native used to take *player* focus into account. The current code seems to no longer to that. The code says

```
            // If we're in play mode and the player has focus (or ignores focus), don't run editor updates.
```

which I think is still what it should do in order to be able to have input in EditorWindow.